openmc/lattice.py

```diff
@@ -408,6 +432,128 @@ def get_cell_instance(self, path, distribcell_index):
 
         return offset
 
+    def find_element(self, point):
+        """Determine index of lattice element and local coordinates for a point
+
+        Parameters
+        ----------
+        point : Iterable of float
+            Cartesian coordinates of point
+
+        Returns
+        -------
+        2- or 3-tuple of int
+            A tuple of the corresponding (x,y,z) lattice element indices
+        3-tuple of float
+            Carestian coordinates of the point in the corresponding lattice
+            element coordinate system
+
+        """
+        ix = floor((point[0] - self._lower_left[0])/self._pitch[0])
+        iy = floor((point[1] - self._lower_left[1])/self._pitch[1])
+        if self.ndim == 2:
+            idx = (ix, iy)
+        else:
+            iz = floor((point[2] - self._lower_left[2])/self._pitch[2])
+            idx = (ix, iy, iz)
+        return idx, self.get_local_coordinates(point, idx)
+
+    def get_local_coordinates(self, point, idx):
+        """Determine local coordinates of a point within a lattice element
+
+        Parameters
+        ----------
+        point : Iterable of float
+            Cartesian coordinates of point
+        idx : Iterable of int
+            (x,y,z) indices of lattice element. If the lattice is 2D, the z
+            index can be omitted.
+
+        Returns
+        -------
+        3-tuple of float
+            Cartesian coordinates of point in the lattice element coordinate
+            system
+
+        """
+        x = point[0] - (self._lower_left[0] + (idx[0] + 0.5)*self._pitch[0])
+        y = point[1] - (self._lower_left[1] + (idx[1] + 0.5)*self._pitch[1])
+        if self.ndim == 2:
+            z = point[2]
+        else:
+            z = point[2] - (self._lower_left[2] + (idx[2] + 0.5)*self._pitch[2])
+        return (x, y, z)
+
+    def get_universe_index(self, idx):
+        """Return index in the universes array corresponding to a lattice element index
```

Comment:
I don't think the difference between the "lattice element index" and "index in the universes array" is clearly defined in the docstrings for these new methods. Can you elaborate a little more in a few of these docstrings?